pkg_resources/__init__.py

```diff
@@ -3323,3 +3321,16 @@ def _initialize_master_working_set():
     # match order
     list(map(working_set.add_entry, sys.path))
     globals().update(locals())
+
+
+#  ---- Ported from ``setuptools`` to avoid introducing dependencies ----
+LOCALE_ENCODING = "locale" if sys.version_info >= (3, 10) else None
+
+
+def _read_utf8_with_fallback(file: str, fallback_encoding=LOCALE_ENCODING) -> str:
+    try:
+        with open(file, "r", encoding="utf-8") as f:
+            return f.read()
+    except UnicodeDecodeError:  # pragma: no cover
+        with open(file, "r", encoding=fallback_encoding) as f:
+            return f.read()
```

Comment:
Maybe emit a warning here so the fallback can be removed?

Reply:
Will we be able to remove this?
The question comes to mind mainly because of `sdist`s that have been produced by old versions of `setuptools` and published to PyPI...